### ISSUE TITLE
SingletonFunction is not supported for the addLambdaFunctions method.

### ISSUE THREAD

**Issue by Davidonium** [DataDog/datadog-cdk-constructs#70]
I have a `SingletonFunction` in my cdk codebase to handle migrations. The typing for the `addLambdaFunctions` method as it is now does not support this construct because it does not inherit from the plain `lambda.Function`, it implements the `lambda.IFunction` interface.

It would be good if `addLambdaFunctions` could support it with:

```typescript
public addLambdaFunctions(
   lambdaFunctions: (lambda.Function | lambdaNodejs.NodejsFunction | lambdaPython.PythonFunction | lambda.SingletonFunction)[],
 )
```

or some similar implementation.

What do you guys think?

**Comment by hghotra** [comment]
After a quick look it seems like `SingletonFunction` is missing some critical properties & methods we need in order to add instrumentation to the Lambda functions.

**Properties:**
```
  public readonly runtime: Runtime;
  public readonly logGroup: logs.ILogGroup;
```

**Methods:**
```
  public addEnvironment(key: string, value: string, options?: EnvironmentOptions): LambdaFunction
  public addLayers(...layers: ILayerVersion[]): void
```

I've created an [issue](https://github.com/aws/aws-cdk/issues/17369) in the `aws-cdk` repo to see if there is a reason these were omitted. If we can add them to the `SingletonFunction` class, then we can go ahead and add support for `SingletonFunction` in `addLambdaFunctions`.

**Comment by hghotra** [comment]
This was fixed and released in [AWS-CDK v1.133](https://github.com/aws/aws-cdk/releases/tag/v1.133.0). Closing the issue now.
